### DISCUSSION TITLE
Error when used in Jest

### DISCUSSION THREAD

**Discussion by insuusvenerati** [andreizanik/cookies-next#28] [Q&A] [answered]
Are there any suggestions for avoiding issues when testing components that get or set cookies?

I'm getting an error that i'm not sure how to address since most solutions involve simply setting the testing environment to jsdom to accommodate `window` and `document`. 

```
/home/stiforr/Developer/hello-free-shavacado/apps/frontend/node_modules/cookies-next/lib/index.js:118
        document.cookie = cookieStr;
                        ^

TypeError: Cannot set properties of null (setting 'cookie')
    at setCookie (/home/stiforr/Developer/hello-free-shavacado/apps/frontend/node_modules/cookies-next/lib/index.js:118:25)
    at /home/stiforr/Developer/hello-free-shavacado/apps/frontend/src/pages/index.tsx:54:92
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by andreizanik** [comment, marked as answer]
Hello 
I think, you need to fully mock the package. You can write your own logic for storing/setting/getting/removing test cookies in a global object.

https://jestjs.io/docs/manual-mocks#mocking-node-modules

**Comment by DonHuskini** [comment]
@insuusvenerati did you find a solution?
